HW-part2

Max-Pochepets opened the pull request:
DAO Part2

Max-Pochepets committed:
intermediate

Max-Pochepets committed:
intermediate

Max-Pochepets committed:
another checkpoint (life is hard)

Max-Pochepets committed:
another checkpoint (life is hard)

Max-Pochepets committed:
Initial

Max-Pochepets committed:
Initial

Max-Pochepets committed:
Merge remote-tracking branch 'origin/HW-part2' into HW-part2

# Conflicts:
#	src/main/java/core/basesyntax/dao/BetDao.java
#	src/main/java/core/basesyntax/dao/Dao.java
#	src/main/java/core/basesyntax/dao/UserDao.java

Max-Pochepets committed:
Refactor:

rebased changes from HW

ivlytvynets commented on src/main/java/core/basesyntax/lib/Injector.java:
```diff
@@ -0,0 +1,33 @@
+                    if (field.getType() == UserDao.class) {
+                        field.set(instance, UserDaoSingleFactory.get());
+                    }
+                }
```
It is said that you should to throw your custom exception if field isn't dao.

Max-Pochepets committed:
Refactor:

Added custom IllegalAnnotationException.java

Max-Pochepets committed:
Refactor:

Single Factory class implemented;
previous factory classes deleted;
Dao interface implemented;
Dao usage generalized;
@Dao -> @DaoAnnotation;

Max-Pochepets committed:
Travis CI

Max-Pochepets committed:
prev state Travis CI

RomanDubovskyi commented on src/main/java/core/basesyntax/lib/DaoAnnotation.java:
```diff
@@ -0,0 +1,8 @@
+import java.lang.annotation.RetentionPolicy;
+
+@Retention(RetentionPolicy.RUNTIME)
+public @interface DaoAnnotation {
```
```suggestion
public @interface Dao {
```

RomanDubovskyi commented on src/main/java/core/basesyntax/lib/Injector.java:
```diff
@@ -0,0 +1,43 @@
+        }
+        if (dao != null && dao.getClass().getAnnotation(DaoAnnotation.class) != null) {
+            return dao;
+        } else {
```
we don't need else after return

Max-Pochepets committed:
@Dao -> @DaoInterface
@DaoAnnotation -> @Dao

RomanDubovskyi commented on src/main/java/core/basesyntax/lib/Dao.java:
```diff
@@ -4,5 +4,5 @@
 
 @Retention(RetentionPolicy.RUNTIME)
-public @interface DaoAnnotation {
+public @interface Dao {
```
missed that you have same name for interface and annotation, my bad 
in this case it would be better to have:
interface - Dao 
annotation - DaoImpl
but for now no need to refactor your code

Max-Pochepets replied:
> missed that you have same name for interface and annotation, my bad
> in this case it would be better to have:
> interface - Dao
> annotation - DaoImpl
> but for now no need to refactor your code

got it, thanks! will keep in mind
